extensions/jms/src/main/java/org/mule/extensions/jms/internal/connection/provider/BaseConnectionProvider.java

```diff
@@ -0,0 +1,314 @@
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extensions.jms.internal.connection.provider;
+
+import static java.lang.String.format;
+import static org.apache.commons.lang.StringUtils.isBlank;
+import static org.mule.extensions.jms.api.connection.JmsSpecification.JMS_1_0_2b;
+import static org.mule.extensions.jms.api.connection.JmsSpecification.JMS_2_0;
+import static org.mule.runtime.api.connection.ConnectionExceptionCode.DISCONNECTED;
+import static org.mule.runtime.api.connection.ConnectionValidationResult.failure;
+import static org.mule.runtime.api.connection.ConnectionValidationResult.success;
+import static org.mule.runtime.core.api.lifecycle.LifecycleUtils.disposeIfNeeded;
+import static org.mule.runtime.core.api.lifecycle.LifecycleUtils.initialiseIfNeeded;
+import static org.mule.runtime.core.api.lifecycle.LifecycleUtils.stopIfNeeded;
+import static org.slf4j.LoggerFactory.getLogger;
+import org.mule.extensions.jms.api.connection.JmsConnection;
+import org.mule.extensions.jms.api.connection.JmsSpecification;
+import org.mule.extensions.jms.api.connection.caching.CachingStrategy;
+import org.mule.extensions.jms.api.connection.caching.DefaultCachingStrategy;
+import org.mule.extensions.jms.api.exception.JmsCallbackConnectionException;
+import org.mule.extensions.jms.internal.connection.JmsCachingConnectionFactory;
+import org.mule.extensions.jms.internal.connection.param.GenericConnectionParameters;
+import org.mule.extensions.jms.internal.support.Jms102bSupport;
+import org.mule.extensions.jms.internal.support.Jms11Support;
+import org.mule.extensions.jms.internal.support.Jms20Support;
+import org.mule.extensions.jms.internal.support.JmsSupport;
+import org.mule.runtime.api.connection.ConnectionException;
+import org.mule.runtime.api.connection.ConnectionValidationResult;
+import org.mule.runtime.api.connection.PoolingConnectionProvider;
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.api.lifecycle.Disposable;
+import org.mule.runtime.api.lifecycle.Initialisable;
+import org.mule.runtime.api.lifecycle.InitialisationException;
+import org.mule.runtime.extension.api.annotation.param.NullSafe;
+import org.mule.runtime.extension.api.annotation.param.Parameter;
+import org.mule.runtime.extension.api.annotation.param.ParameterGroup;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import javax.jms.Connection;
+import javax.jms.ConnectionFactory;
+import javax.jms.ExceptionListener;
+import javax.jms.JMSException;
+
+import org.slf4j.Logger;
+
+/**
+ * Base implementation of a {@link PoolingConnectionProvider} for {@link JmsConnection}s
+ *
+ * @since 4.0
+ */
+public abstract class BaseConnectionProvider implements PoolingConnectionProvider<JmsConnection>, Initialisable, Disposable {
```

Comment:
Please verify that all connections to JMS are created during start() and not initialise()